### ISSUE TITLE
How to prevent collision

### ISSUE THREAD

**Issue by celikemir** [unr-arl/gbplanner_ros#53]
Hi,

I have a somewhat challenging map where the elevation changes. In some areas, the elevation from the ground is really small, like 5 cm or below. How can I prevent the planner from drawing paths in these kinds of places? There are pavements, and the planner draws above the pavements, which it should not do. How can I prevent this behavior? I modified col_check = 2 in the CMakeLists file, which resulted in some improvement but didn't fully solve the problem.
![soru](https://github.com/ntnu-arl/gbplanner_ros/assets/33879396/a08ff73b-0cd6-423b-904a-e66294579cee)

**Comment by MihirDharmadhikari** [comment]
Hello,

The pavements are tricky to tackle. Due to the size of the voxels in the voxblox map, the planner won't be able to differentiate between a slope and a pavement that is one voxel size tall. One workaround to this is to reduce the voxel size (currently, it is 0.2 you can make it 0.1m). However, keep in mind that this will increase the memory significantly as well as increase the planning time.

Best,
Mihir

**Comment by celikemir** [comment]
thank you Mihir. The global area, the red sqaure, is not cover all my map so I have modify the gbplanner.yaml [there](https://github.com/ntnu-arl/gbplanner_ros/blob/77afce521a3aa7a4b94bf6a43a3c88ec08b6d544/gbplanner/config/smb/gbplanner_config.yaml#L34) is a defined number even though I changed so small or large number cant see difference dimension of the red box.

**Comment by celikemir** [comment]
yeah it brings realy huge memory consumption. maybe you can adress me reduce voxblock count I tried to reduce with some yaml[https://github.com/ntnu-arl/gbplanner_ros/blob/77afce521a3aa7a4b94bf6a43a3c88ec08b6d544/gbplanner/config/smb/voxblox_sim_config.yaml#L4](url)

Can you tell me some of the parameters meaning ? thank you

**Comment by celikemir** [comment]
@MihirDharmadhikari How to increase the per voxel block size to prevent collision? There is a huge difference between the planned path height and the obstacle. However, the robot is supposed to go over the obstacle, but it overturns. The robot also came closer to the obstacle, though it had been seen before. Then it tried to replan. how you can solve it.

and why did you remove  PlanningParams/planning_backward: If the path is to be executed backwards.

param ?


This path is funny ?
![Screenshot from 2024-06-12 18-25-37](https://github.com/ntnu-arl/gbplanner_ros/assets/33879396/6b65fe17-a6ab-4d33-8856-6ff2cb81394c)


Can you please explain what causes for this problem 
![Screenshot from 2024-06-12 18-30-16](https://github.com/ntnu-arl/gbplanner_ros/assets/33879396/e17ee768-c100-4911-af2d-7e7a100f037d)
